### DISCUSSION TITLE
Deploy > managed-learn > ml-project-service

### DISCUSSION THREAD

**Discussion by sandeep-r1** [Sunbird-Ed/Community#531] [Bugs]
Hi Team,
We are facing issue while deploying ml-project-service. Please find the error below. Kindly help us to resolve this issue.
```
TASK [sunbird-deploy : helm install] *******************************************
fatal: [localhost]: FAILED! => {
    "changed": true,
    "cmd": "helm upgrade --install --atomic --timeout 10m ml-projects-service /var/lib/jenkins/workspace/Deploy/alex/dev/managed-learn/ml-projects-service/kubernetes/helm_charts/core/ml-projects-service -n alexdev",
    "delta": "0:10:00.655924",
    "end": "2023-05-30 15:15:26.362117",
    "rc": 1,
    "start": "2023-05-30 15:05:25.706193"
}

STDOUT:

Release "ml-projects-service" does not exist. Installing it now.


STDERR:

Error: release ml-projects-service failed, and has been uninstalled due to atomic being set: timed out waiting for the condition


MSG:

non-zero return code


PLAY RECAP *********************************************************************
localhost                  : ok=16   changed=14   unreachable=0    failed=1    skipped=5    rescued=0    ignored=0
```

**Comment by tsprasath** [comment]
@sandeep-r1  Please share the pods logs.

**Reply by srikanth-6520** [reply]
@tsprasath we have provided pod logs of the service below.

 APPLICATION_PORT            │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ APPLICATION_ENV             │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ MONGODB_URL                 │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ INTERNAL_ACCESS_TOKEN       │ Required internal access token │
├─────────────────────────────┼────────────────────────────────┤
│ KAFKA_COMMUNICATIONS_ON_OFF │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ KAFKA_URL                   │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ USER_SERVICE_URL            │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ SERVICE_NAME                │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ CERTIFICATE_SERVICE_URL     │ PASSED                         │
├─────────────────────────────┼────────────────────────────────┤
│ PROJECT_CERTIFICATE_ON_OFF  │ PASSED                         │
└─────────────────────────────┴────────────────────────────────┘
issuer Kid url :  http://registry-service:8081/api/v1/PublicKey/search
issuer Kid bodyData :  {"filters":{}}
Environment : alexdev
Application is running on the port : 3000
Connected to DB
Connected to Kafka
KID success response :  "upstream connect error or disconnect/reset before headers. reset reason: connection failure, transport failure reason: delayed connect error: 111"
Server stoped . Failed to set certificate issuer Kid value
Stream closed EOF for alexdev/ml-projects-service-784db95d76-4hbc2 (ml-projects-service)

**Reply by tsprasath** [reply]
[aks30](https://github.com/aks30) Kindly help @srikanth-6520 on ML service

**Reply by chitranshu-keshav** [reply]
@aks30 @Prateek-slokam @kiranharidas187 Can you please get back on this issue

**Reply by aks30** [reply]
@srikanth-6520 Is the registry service up and running? Project service requires that service as a dependency, if that service is not available you can set PROJECT_CERTIFICATE_ON_OFF value as OFF in which case it will not look for that dependency.

**Reply by chitranshu-keshav** [reply]
@srikanth-6520 is your issue resolved? If yes then can you mark this thread as answered

**Comment by sandeep-r1** [comment]
Answered
